controller/src/main/java/io/pravega/controller/store/PravegaTablesScope.java

```diff
@@ -218,7 +218,8 @@ public void refresh() {
 
     public CompletableFuture<Void> addReaderGroupToScope(String readerGroupName, UUID readerGroupId) {
         return getReaderGroupsInScopeTableName()
-                .thenCompose(tableName -> Futures.toVoid(storeHelper.addNewEntry(tableName, readerGroupName, getIdInBytes(readerGroupId))));
+                .thenCompose(tableName -> storeHelper.createTable(tableName).thenCompose(v ->
```

Comment:
Is `createTable` an idempotent operation? 

Or will it go to the Segment Store and try to create a table, thus flooding it with create table requests which will fail most of the time. That will create unnecessary churn and logs.

Perhaps a better approach would be to add an exception handler (`Futures.exceptionallyComposeExpecting` ....) which creates the table and retries if it gets an exception that the table did not exist in the first place (i.e., optimistic approach).

Reply:
`createTable` is an idempotent operation but it would hit the segment store as you suggested.
Will add it the `Futures.exceptionallyComposeExpecting` way.

Reply:
fixed as suggested.